unlock-protocol.com/blog/emergency-upgrade-1-3.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Emergency Upgrade
+subTitle: Version 1-3 is now available
+authorName: Nick Mancuso
+publishDate: January 17, 2020
+description: Explaining what went wrong and why we released a new version so quickly after our last upgrade.
+image: /static/images/blog/emergency-upgrade-1-3/code_vulnerability.jpg
+---
+
+We recently noticed a vulnerability which if abused could have had bad consequences – but we caught it early and fixed it promptly. This post aims to explain the errors we made with our [most recent smart-contract update](https://unlock-protocol.com/blog/version-1-2/).
+
+There were problems with this release which if targeted by an attacker had the potential to do one of the following:
+ - Steal money made from previous key sales.
+ - Break every lock (and key) using the new version - blocking new sales, preventing key owners from being recognized and preventing the lock owner from accessing any funds previously made.
+
+We fixed the second issue very quickly, so no worries there.  But funds are at risk if your lock is impacted.  For that we need the owners of these locks themselves to upgrade ASAP. (there are 14 locks which affected)
+
+This only impacts locks created between Jan 6th until Jan 17th (locks created in blocks #9227829 - #9295958). If you recently created a lock, please go to the Unlock dashboard to check if you are impacted. If the dashboard does not show a warning like the one below, your lock is safe.
+
+<p style="text-align:center">
+	<img src="/static/images/blog/emergency-upgrade-1-3/dashboard_warning.png" width="750px" alt="Dashboard warning">
+</p>
+
+So what happened?  Let’s start with the issue we were able to fix right away and then we’ll get to the juicy stuff.
+
+## Break every lock using the new version
+
+> This issue is about our deployment process.  We simply needed to take action before an attacker.  We will avoid this race condition in the future.
+
+We recently changed the locks deployed to use the minimal proxy standard, [EIP-1167](https://eips.ethereum.org/EIPS/eip-1167).  This significantly reduces the cost to create new locks and leads to less blockchain bloat.  When using proxies any data set in the constructor is lost so instead we call a single-use function called `initialize`.
+
+During normal operation a user calls `createLock` on our main Unlock contract.  That function will deploy the lock and call `initialize` in the same transaction.  It’s responsible for recording data such as the lock owner and the key price.
+
+Proxies work by reading the implementation logic from a template.  The template is nothing more than a contract containing the logic / source code for the lock proxies to leverage.
```

Comment:
```suggestion
Proxies work by reading the implementation logic from a template.  The template is nothing more than a contract containing the logic / source code for the lock implementations to leverage.
```

Reply:
hmm that sounds a little confusing to me.  maybe `... / source code for each individual lock to leverage`

Reply:
I pushed with the text I suggested.  let me know if this sounds alright.